data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/S3Source.java

```diff
@@ -33,12 +35,23 @@ public class S3Source implements Source<Record<Event>> {
     private final S3SourceConfig s3SourceConfig;
     private SqsService sqsService;
     private final PluginFactory pluginFactory;
+    private final EventFactory eventFactory;
+    private final AcknowledgementSetManager acknowledgementSetManager;
+    private final boolean endToEndAcknowledgementsEnabled;
 
     @DataPrepperPluginConstructor
-    public S3Source(PluginMetrics pluginMetrics, final S3SourceConfig s3SourceConfig, final PluginFactory pluginFactory) {
+    public S3Source(PluginMetrics pluginMetrics, final S3SourceConfig s3SourceConfig, final PluginFactory pluginFactory, final EventFactory eventFactory, final AcknowledgementSetManager acknowledgementSetManager) {
         this.pluginMetrics = pluginMetrics;
+        this.eventFactory = eventFactory;
         this.s3SourceConfig = s3SourceConfig;
         this.pluginFactory = pluginFactory;
+        this.endToEndAcknowledgementsEnabled = s3SourceConfig.getEndToEndAcknowledgements();
+        this.acknowledgementSetManager = acknowledgementSetManager;
+    }
+
+    @Override
+    public boolean areAcknowledgementsEnabled() {
+        return endToEndAcknowledgementsEnabled;
```

Comment:
Question: is this required? I don't see it being used.

Reply:
It used via the source interface in `./data-prepper-core/src/main/java/org/opensearch/dataprepper/parser/PipelineParser.java`